pkg/scalers/arangodb_scaler.go

```diff
@@ -0,0 +1,276 @@
+		if err != nil {
+			return nil, fmt.Errorf("failed to create bearer token authorization header, %v", err)
+		}
+		auth = driver.RawAuthentication(hdr)
+	}
+
+	client, err := driver.NewClient(driver.ClientConfig{
+		Connection:     conn,
+		Authentication: auth,
+	})
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize a new client, %v", err)
+	}
+
+	return client, nil
+}
+
+func parseArangoDBMetadata(config *ScalerConfig) (*arangoDBMetadata, error) {
+	var err error
+
+	// setting default metadata
+	meta := arangoDBMetadata{}
+
+	// parse metaData from ScaledJob config
+	meta.endpoints, err = GetFromAuthOrMeta(config, "endpoints")
+	if err != nil {
+		return nil, err
+	}
+
+	if val, ok := config.TriggerMetadata["collection"]; ok {
+		meta.collection = val
+	} else {
+		return nil, fmt.Errorf("no collection given")
+	}
+
+	if val, ok := config.TriggerMetadata["query"]; ok {
+		meta.query = val
+	} else {
+		return nil, fmt.Errorf("no query given")
+	}
+
+	if val, ok := config.TriggerMetadata["queryValue"]; ok {
+		queryValue, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert queryValue to int, %v", err)
+		}
+		meta.queryValue = queryValue
+	} else {
+		return nil, fmt.Errorf("no queryValue given")
+	}
+
+	meta.activationQueryValue = 0
+	if val, ok := config.TriggerMetadata["activationQueryValue"]; ok {
+		activationQueryValue, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert activationQueryValue to int, %v", err)
+		}
+		meta.activationQueryValue = activationQueryValue
+	}
+
+	meta.dbName, err = GetFromAuthOrMeta(config, "dbName")
+	if err != nil {
+		return nil, err
+	}
+
+	meta.unsafeSsl = false
+	if val, ok := config.TriggerMetadata["unsafeSsl"]; ok && val != "" {
+		unsafeSslValue, err := strconv.ParseBool(val)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse unsafeSsl, %v", err)
+		}
+		meta.unsafeSsl = unsafeSslValue
+	}
+
+	if val, ok := config.TriggerMetadata["connectionLimit"]; ok {
+		connectionLimit, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("failed to convert connectionLimit to int, %v", err)
+		}
+		meta.connectionLimit = connectionLimit
+	}
+
+	// parse auth configs from ScalerConfig
+	meta.arangoDBAuth, err = authentication.GetAuthConfigs(config.TriggerMetadata, config.AuthParams)
+	if err != nil {
+		return nil, err
+	}
+
+	meta.scalerIndex = config.ScalerIndex
+	return &meta, nil
+}
+
+func (s *arangoDBScaler) IsActive(ctx context.Context) (bool, error) {
+	result, err := s.getQueryResult(ctx)
+	if err != nil {
+		s.logger.Error(err, fmt.Sprintf("failed to get query result by arangoDB, %v", err))
+		return false, err
+	}
+	return result > s.metadata.activationQueryValue, nil
+}
+
+// Close disposes of arangoDB connections
+func (s *arangoDBScaler) Close(ctx context.Context) error {
+	return nil
```

Comment:
Should we close the client? (IDK if it's possible and/or required)

Reply:
It isn't required since the client connection isn't persistent. The client is simply a wrapper around the HTTP REST API, which makes transient API calls by default.